twopaneview/TwoPaneView.js

```diff
@@ -0,0 +1,215 @@
+    Dimensions.addEventListener('change', this._handleDimensionsChange);
+  }
+
+  componentWillUnmount() {
+    Dimensions.removeEventListener('change', this._handleDimensionsChange);
+  }
+
+  _handleDimensionsChange = dimensions => {
+    this.setState({
+      dims: dimensions.window,
+      spanning: DualScreenInfo.isSpanning 
+    });
+    this.state.onModeChanged();
+  };
+
+  render() {    
+
+    let direction ='row';
+
+    return (
+      <View style={{flexDirection: direction, width: this.state.dims.width, height:this.state.dims.height}}>
+        {this.renderChildPanes()}
+      </View>
+    );
+  }
+
+  renderChildPanes() {
+    const children = React.Children.toArray(this.props.children);
+
+    if (this.state.spanning) {
+      if (this.state.paneMode === PaneMode.Single || this.state.dims.height > this.state.dims.width  && this.state.panePriorityVerticalSpanning) {
+        if (this.state.panePriorityVerticalSpanning === TPV_PanePriority.Pane1) {
+          return this.renderPane1(this.getEntireSize());
+        }
+        else {
+          return this.renderPane2(this.getEntireSize());
+        }
+      }
+      return this.renderBothPanes();
+    }
+    if (this.state.paneMode === PaneMode.Dual) {
+      return this.renderDualPanes();
+    }
+    if (this.state.panePriority === TPV_PanePriority.Pane1) {
+      return this.renderPane1(this.getEntireSize());
+    }
+    return this.renderPane2(this.getEntireSize());
+  }
+
+  renderDualPanes() {
+    const children = React.Children.toArray(this.props.children);
+
+    const items = [];
+    if (children.length > 0) {
+      items.push(this.renderPane1(this.getLeftSize()));
+    }
+
+
+    if (children.length > 1) {
+      items.push(this.renderPane2(this.getRightSize()));
+    }
+
+    return items;
+  }
+
+  renderBothPanes() {
+    let horizontal = this.state.dims.width >= this.state.dims.height;
+    const children = React.Children.toArray(this.props.children);
+
+    const items = [];
+    if (children.length > 0) {
+      items.push(this.renderPane1(horizontal ? this.getLeftSize() : this.getTopSize()));
+    }
+
+    items.push(this.renderSeparator());
+
+    if (children.length > 1) {
+      items.push(this.renderPane2(horizontal ? this.getRightSize() : this.getBottomSize()));
+    }
+
+    return items;
+  }
+
+  renderPane1(size) {
+    const children = React.Children.toArray(this.props.children);
+    if (children.length > 0) {
+      return (
+        <View key="pane1" style={{width:size.width, height:size.height}}>
+          {children[0]}
+        </View>
+      );
+    }
+  }
+
+  renderPane2(size) {
+    const children = React.Children.toArray(this.props.children);
+    if (children.length > 1) {
+      return (
+        <View key="pane2" style={{width:size.width, height:size.height}}>
+          {children[1]}
+        </View>
+      );
+    }
+  }
+
+  renderSeparator() {
+    let horizontal = this.state.dims.width >= this.state.dims.height;
```

Comment:
>let horizontal = this.state.dims.width >= this.state.dims.height; [](start = 4, length = 65)

This isn't a reliable way to detect the orientation of the device.  We can discuss offline.

Reply:
You are right and personally I do not like this hack, but could not find a better way to detect this. Please, I am open to better suggestions if you now of a better way to detect orientation.

Reply:
I can think of a couple ways to detect the orientation:
1) It can be deduced from the windowRects coming from DualScreenInfo.  If the device is horizontal, windowRects[1] will have a left that is != windowRects[0], and similarly if it's vertical, windowRects[1] will have a top that's != windowRects[0].  Of course there will only be two windowRects in the array when running on a dual-screen device, but this info should only be needed in that scenario.
2) There's a community module called react-native-orientation.  I think I would prefer option 1 to avoid another dependency, but I thought I should mention it.

You can fix this later BTW as what you have will work for all known devices at this point in time.  If you decide to fix it later, just file an issue and we can track it.